### PR TITLE
fix: sass config

### PR DESCRIPTION
**Pull request by tomleo**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Update vite to [5.4.0](https://github.com/vitejs/vite/blob/v5.4.0/packages/vite/CHANGELOG.md#540-2024-08-07) which fixes a [bug](https://github.com/vitejs/vite/pull/17754) via https://github.com/vitejs/vite/commit/1025bb6d8f21c0cb9fe72405d42e0f91bb3f1d8e allowing hooks into the dart-sass API to silence deprecation warnings
- Update to silence deprecation warnings (also applied to `es-bs-base`)
- Made sure we're using the same version of SASS in `es-bs-base` and `es-ds-docs`

#### 🧐 Feedback Requested / Focus Areas

-

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [ ] I have documented testing approach

### CODE REVIEW

File: es-bs-base/package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
   "style": "dist/css/bootstrap.css",
   "scripts": {
     "css": "npm-run-all css-compile css-prefix css-minify",
-    "css-compile": "sass scss/:dist/css/",
+    "css-compile": "sass --silence-deprecation=mixed-decls scss/:dist/css/",
```
Comment (tomleo) [inline es-bs-base/package.json:34]: This argument silences mixed-decls deprecation

File: es-ds-docs/nuxt.config.ts [hunk 2]
```diff
@@ -7,6 +7,19 @@ export default defineNuxtConfig({
         '@energysage/es-bs-base/scss/bootstrap.scss',
     ],
 
+    // The following will still not work until https://github.com/alibaba/formily/pull/4195
+    vite: {
+        css: {
+            preprocessorOptions: {
+                scss: {
+                    silenceDeprecations: [
+                        'mixed-decls', // https://sass-lang.com/documentation/breaking-changes/mixed-decls/
```
Comment (tomleo) [inline es-ds-docs/nuxt.config.ts:16]: This passes the same argument to vite that was passed in `es-bs-base` above to silence in development, however there's a pending PR upstream that must be merged before this argument will work

**Review by nathanielwarner — COMMENTED** [review]
I think we should [try fixing the deprecation warnings](https://energysage.atlassian.net/browse/CED-1731) before we silence them. If we ignore them, the behavior will change in a future version of sass.

**Comment by tomleo** [comment]
> I think we should [try fixing the deprecation warnings](https://energysage.atlassian.net/browse/CED-1731) before we silence them. If we ignore them, the behavior will change in a future version of sass.

The particular warning being silenced was

```
175 │ ┌           > .container#{breakpoints.breakpoint-infix($size, variables.$container-max-widths)} {
176 │ │             @extend %container-navbar-expand-#{$breakpoint};
177 │ │           }
    │ └─── nested rule
... │
183 │           justify-content: flex-start;
    │           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
    ╵
    ../es-bs-base/scss/_navbar.scss 183:9             @content
    ../es-bs-base/scss/mixins/_breakpoints.scss 69:5  media-breakpoint-up()
    ../es-bs-base/scss/_navbar.scss 181:7             @use
    ../es-bs-base/scss/bootstrap.scss 27:1            root stylesheet
```

**Comment by nathanielwarner** [comment]
> > I think we should [try fixing the deprecation warnings](https://energysage.atlassian.net/browse/CED-1731) before we silence them. If we ignore them, the behavior will change in a future version of sass.
> 
> The particular warning being silenced was
> 
> ```
> 175 │ ┌           > .container#{breakpoints.breakpoint-infix($size, variables.$container-max-widths)} {
> 176 │ │             @extend %container-navbar-expand-#{$breakpoint};
> 177 │ │           }
>     │ └─── nested rule
> ... │
> 183 │           justify-content: flex-start;
>     │           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
>     ╵
>     ../es-bs-base/scss/_navbar.scss 183:9             @content
>     ../es-bs-base/scss/mixins/_breakpoints.scss 69:5  media-breakpoint-up()
>     ../es-bs-base/scss/_navbar.scss 181:7             @use
>     ../es-bs-base/scss/bootstrap.scss 27:1            root stylesheet
> ```

That's correct yeah. This is the issue in question I think we should try to address. It just involves moving the declaration above the nested rule for each of the 6 instances

**Pull request closed without merging**